server/helpers/authHelpers.js

```diff
@@ -0,0 +1,126 @@
+      {
+        fields: [
+          'lastname', 'firstname', 'email', 'password', 'isVerified'
+        ]
+      });
+    return result;
+  }
+
+  /**
+   * This method handle the signup request.
+   * @param {String} email the email to send the confirmation to.
+   * @param {String} token the token to assign to the email
+   * @returns {object}  some data of the created user from database.
+   */
+  static async sendMail(email, token) {
+    const url = `http://localhost:3000/api/v1/auth/verify/${token}`;
+    const msg = {
+      to: email,
+      from: 'champsdev2@gmail.com',
+      subject: 'Comfirmation email',
+      text: 'welcome to nomad champs',
+      html: `
+      <div style="background-color: white;border: wheat 2px solid;padding: 20px;
+      max-width: 50vw;margin: 50px;align-items: center;height: 45vh;display: flex; flex-direction: column;">
+          <h1 style="font-size: 25px;margin-top: 30px;">Welcome to Nomad champs</h1>
+          <p style="font-family: fantasy;font-size: 20px;">Comfirm your email to proceed to Nomad champs</p>
+          <button style="background-color: rgb(11, 132, 212);
+          width: 130px;height: 30px;border: none;margin-top: 40px;
+          "><a href="${url}" style="text-decoration: none;color: white;font-size: 14px;
+          ">
+          Confirm Email</a>
+          </button>
+      </div>
+      `,
+    };
+    try {
+      sgMail.send(msg);
+      return { isSent: true };
+    } catch (error) {
+      return {
+        isSent: false,
+        error
+      };
+    }
+  }
+
+  /**
+   * This method handle the signup request.
+   * @param {object} data the object that contain the data that is embedded with token.
+   * @returns {String}  token created .
+   */
+  static async createToken(data) {
+    try {
+      const token = jwt.sign(data,
+        process.env.JWT_KEY,
+        { expiresIn: '2h' });
+      return token;
+    } catch (error) {
+      return error;
+    }
+  }
+
+  /**
+   * This method handle the signup request.
+   * @param {String} token the string that contain a token.
+   * @returns {object}  some data of the created user from database.
+   */
+  static verifyToken(token) {
+    try {
+      const data = jwt.verify(token, process.env.JWT_KEY);
+      return data;
+    } catch (err) {
+      return err;
+    }
+  }
+
+  /**
+   * This method handle the signup request.
+   * @param {Number} id the id of the user to be verified.
+   * @returns {Boolean}  if the data were created or not.
+   */
+  static async verify(id) {
+    const verify = await User.update({ isVerified: true }, { where: { id } });
+    if (verify) {
+      return true;
+    // eslint-disable-next-line no-else-return
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * This method handle the signup request.
+   * @param {String} email the email of the user to be verified.
+   * @returns {Boolean}  if the user exists or not.
+   */
+  static async isUser(email) {
```

Comment:
Consider updating the function name to **findUser** for meaningfulness.

Reply:
I already resolved that!!!